### PR TITLE
Partial implementation for XProcessing with KSP

### PR DESCRIPTION
**Pull request by yigit**
XType implementation with KSP

This PR adds the first bits of support for KSP in Room. It is still far from complete but moving development back to aosp since both AndroidX and KSP moved to Kotlin 1.4.0

Bug: 160322705
Test: KspTypeTest,  KspTypeExtTest

### CODE REVIEW

**Comment by yigit** [comment]
@danysantiago this is combination of CLs you've already reviewed in my fork.
Once it is mirrored into AOSP, i'll add prebuilts.

**Comment by yigit** [comment]
try again github

**Review by danysantiago — APPROVED** [review]

File: buildSrc/src/main/kotlin/androidx/build/AndroidXPlaygroundRootPlugin.kt [hunk 1]
```diff
@@ -157,6 +157,10 @@ class AndroidXPlaygroundRootPlugin : Plugin<Project> {
         google()
         mavenCentral()
         jcenter()
+        // TODO temporary until AndroidX merge
+        maven {
+            it.url = URI("https://jitpack.io")
```
Comment (danysantiago) [inline buildSrc/src/main/kotlin/androidx/build/AndroidXPlaygroundRootPlugin.kt]: Probably need to remove this?

Comment (yigit) [reply buildSrc/src/main/kotlin/androidx/build/AndroidXPlaygroundRootPlugin.kt]: good catch.
I also need to move away from its snapshot to an actual release but that is breaking the build right now, will take a look but may need to wait for another release from kotlin compile testing.

**Pull request closed without merging**